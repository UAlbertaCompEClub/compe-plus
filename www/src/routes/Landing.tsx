// Libraries
import React, { FC } from 'react';
import { useAuth0 } from '@auth0/auth0-react';

// Material UI
import { makeStyles } from '@material-ui/styles';
import { Grid, Typography, Box, Button } from '@material-ui/core';

// Assets
import PairProgramming from '../assets/pair_programming.svg';
import LandingBackground from '../assets/LandingBackground.svg';
import BlackLogo from '../assets/logo_black.svg';
import Practice from '../assets/practice.svg';
import MockInterview from '../assets/mock_interview.svg';
import Industry from '../assets/industry.svg';
import ResumeReview from '../assets/resume_review.svg';
import Teamwork from '../assets/teamwork.svg';
import Support from '../assets/support.svg';

const Landing: FC = () => {
    return (
        <Grid container direction='row' justify='center' style={{ backgroundImage: `url(${LandingBackground})`, backgroundSize: 'cover', paddingBottom: 80 }}>
            <Grid container item spacing={10} xs={10}>
                <Intro />
                <Values />
                <Services />
            </Grid>
        </Grid>
    );
};

export const Intro: FC = () => {
    const classes = useStyles();
    const { isAuthenticated } = useAuth0();

    return (
        <Grid container item>
            <Grid container item style={{ height: '100vh' }}>
                <Grid container item xs={5} alignItems='center'>
                    <Grid item>
                        <Box maxWidth='sm'>
                            <img src={BlackLogo} className={classes.logo_1} />
                        </Box>
                        <Typography variant='h1'>CompE+</Typography>
                        <Typography variant='h3'>Practice, Insight, and Support</Typography>
                    </Grid>
                </Grid>
                <Grid container item xs={7} alignItems='center' justify='center'>
                    <Box maxWidth='sm'>
                        <img src={PairProgramming} className={classes.svg} />
                    </Box>
                </Grid>
                <Grid container item xs={12} justify='center'>
                    <Grid container item alignItems='center' justify='center' direction='column' spacing={3}>
                        <Grid item>
                            <Typography component='h2' variant='h2' align='center'>
                                CompE+ is a student-led pilot program to provide support for computer engineering students looking for internships
                            </Typography>
                        </Grid>
                        {!isAuthenticated && (
                            <Grid item>
                                <Button variant='contained' color='primary' onClick={() => console.log('clicked Get Started button')}>
                                    {'Get Started'}
                                </Button>
                            </Grid>
                        )}
                    </Grid>
                </Grid>
            </Grid>
            <Grid container item spacing={10}>
                <Grid container item xs={8} alignItems='center' justify='center'>
                    <Grid item>
                        <Typography component='h2' variant='h1' noWrap>
                            What’s CompE+ ?
                        </Typography>
                        <Typography component='h2' variant='body1'>
                            CompE+ is a student led pilot program to provide CompE students with practice, insight, and support when finding internships in the tech industry. Arguably, computer
                            engineering is a unique discipline and requires different resources and support - that’s where CompE+ comes in!{' '}
                        </Typography>
                    </Grid>
                </Grid>
                <Grid container item xs={4} alignItems='center' justify='center'>
                    <img src={BlackLogo} className={classes.logo_2} />
                </Grid>
            </Grid>
        </Grid>
    );
};

const Values: FC = () => {
    const classes = useStyles();

    return (
        <Grid container item spacing={8}>
            <Grid container item>
                <Typography variant='h1'>Values</Typography>
            </Grid>
            <Grid container item spacing={5}>
                <Grid container item xs={3} className={classes.values}>
                    <img src={Practice} className={classes.svg} />
                </Grid>
                <Grid item xs={9}>
                    <Typography variant='h2'>Practice</Typography>
                    <Typography variant='body1'>
                        Technical interviews can be difficult and requires a good amount of practice. By providing resources and services, CompE+ can help computer engineering students tackle the
                        technical interview with greater confidence
                    </Typography>
                </Grid>
            </Grid>
            <Grid container item spacing={5}>
                <Grid item xs={9}>
                    <Typography align='right' variant='h2'>
                        Insight
                    </Typography>
                    <Typography align='right' variant='body1'>
                        The tech industry can be a black box to new students. CompE+ strives to provide insight into the industry by sharing the internship experience with all computer engineering
                        students
                    </Typography>
                </Grid>
                <Grid container item xs={3} className={classes.values}>
                    <img src={Industry} className={classes.svg} />
                </Grid>
            </Grid>
            <Grid container item spacing={5}>
                <Grid container item xs={3} className={classes.values}>
                    <img src={Support} className={classes.svg} />
                </Grid>
                <Grid item xs={9}>
                    <Typography variant='h2'>Support</Typography>
                    <Typography variant='body1'>
                        Entering the tech industry can be intimidating to many, even after landing the first internship, imposter syndrome can hit hard. Fostering a positive community is an important
                        goal of CompE+
                    </Typography>
                </Grid>
            </Grid>
        </Grid>
    );
};

const Services: FC = () => {
    const classes = useStyles();

    return (
        <Grid container item spacing={8}>
            <Grid container item>
                <Typography variant='h1'>Services</Typography>
            </Grid>
            <Grid container item spacing={2}>
                <Grid container item xs={12} spacing={3}>
                    <Grid container item xs={4} justify='center'>
                        <img src={ResumeReview} className={classes.services_svg} />
                    </Grid>
                    <Grid container item xs={4} justify='center'>
                        <img src={MockInterview} className={classes.services_svg} />
                    </Grid>
                    <Grid container item xs={4} justify='center'>
                        <img src={Teamwork} className={classes.services_svg} />
                    </Grid>
                </Grid>

                <Grid container item xs={12} spacing={3}>
                    <Grid container item xs={4} justify='center'>
                        <Typography variant='h2'>Resume Review</Typography>
                    </Grid>
                    <Grid container item xs={4} justify='center'>
                        <Typography variant='h2'>Mock Interviews</Typography>
                    </Grid>
                    <Grid container item xs={4} justify='center'>
                        <Typography variant='h2'>Community</Typography>
                    </Grid>
                </Grid>

                <Grid container item xs={12} spacing={3}>
                    <Grid item xs={4}>
                        <Typography align='center' variant='body1'>
                            Led by senior CompE students, practice communicating your thoughts and problem-solving skills with mock technical interviews
                        </Typography>
                    </Grid>
                    <Grid item xs={4}>
                        <Typography align='center' variant='body1'>
                            Simulate the application process by getting your resume reviewed by the co-op office followed by a senior compE student to provide a technical perspective
                        </Typography>
                    </Grid>
                    <Grid item xs={4}>
                        <Typography align='center' variant='body1'>
                            Connect with other CompE students, gain insight on different companies that other students have worked for, and learn more about the tech industry
                        </Typography>
                    </Grid>
                </Grid>
            </Grid>
        </Grid>
    );
};

const useStyles = makeStyles(() => ({
    svg: {
        width: '100%',
        height: 'auto',
    },
    values: {
        justifyContent: 'center',
        alignItems: 'center',
    },
    services_svg: {
        width: '80%',
        height: 'auto',
    },
    logo_2: {
        width: '70%',
        height: 'auto',
    },
    logo_1: {
        width: '50%',
        height: 'auto',
    },
}));

export default Landing;
