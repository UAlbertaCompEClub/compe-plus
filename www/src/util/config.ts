const AUTH0_DOMAIN = process.env.REACT_APP_AUTH0_DOMAIN || '';
const AUTH0_CLIENT_ID = process.env.REACT_APP_AUTH0_CLIENT_ID || '';
const SERVER_ENDPOINT = process.env.REACT_APP_SERVER_ENDPOINT || 'http://localhost:1337';
const SERVER_AUDIENCE = process.env.REACT_APP_SERVER_AUDIENCE || '';
const MAILCHIMP_URL = process.env.REACT_APP_MAILCHIMP_URL || '';

const AUTH0 = {
    domain: AUTH0_DOMAIN,
    clientId: AUTH0_CLIENT_ID,
};

const SERVER = {
    endpoint: SERVER_ENDPOINT,
    audience: SERVER_AUDIENCE,
};

const config = {
    auth0: AUTH0,
    server: SERVER,
    mailchimpUrl: MAILCHIMP_URL,
};

export default config;
