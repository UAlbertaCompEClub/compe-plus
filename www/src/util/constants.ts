export const COMPE_PLUS = 'CompE+';

export const ABOUT = 'About';
export const RESUME_REVIEW = 'Resume Review';
export const MOCK_INTERVIEW = 'Mock Interview';
export const COMMUNITY = 'Community';
export const COMPE_CLUB = 'CompE Club';

// Routes
export const RESUME_REVIEW_ROUTE = '/resume-review';
export const MOCK_INTERVIEW_ROUTE = '/mock-interview';
export const COMMUNITY_ROUTE = '/community';
export const REGISTER_ROUTE = '/register';
export const COMPE_CLUB_ROUTE = 'https://www.compeclub.com/';
