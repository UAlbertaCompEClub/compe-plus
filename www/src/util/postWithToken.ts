import { useAuth0 } from '@auth0/auth0-react';
import axios, { AxiosResponse } from 'axios';

export const postWithToken = async <DataType, ResponseType>(url: string, scopes: string[], data?: DataType): Promise<AxiosResponse<ResponseType> | undefined> => {
    const { getAccessTokenSilently } = useAuth0();
    try {
        const token = await getAccessTokenSilently({
            scope: scopes.join(' '),
        });
        return axios.post(url, data, {
            headers: {
                authorization: `Bearer ${token}`,
            },
        });
    } catch (e) {
        console.error(e);
    }
    return undefined;
};
