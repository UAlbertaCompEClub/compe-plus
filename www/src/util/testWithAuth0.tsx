/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { Auth0Context, Auth0ContextInterface } from '@auth0/auth0-react';
import { render, RenderResult } from '@testing-library/react';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

function shallowWithAuth0(
    component: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
    authState: Partial<Auth0ContextInterface>,
): Enzyme.ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>> {
    return shallow(component, {
        wrappingComponent: Auth0Context.Provider,
        wrappingComponentProps: {
            value: authState,
        },
    });
}

function renderWithAuth0(component: React.ReactElement<any, string | React.JSXElementConstructor<any>>, authState: Partial<Auth0ContextInterface>): RenderResult {
    return render(<Auth0Context.Provider value={authState as any}>{component}</Auth0Context.Provider>);
}

export { renderWithAuth0, shallowWithAuth0 };
