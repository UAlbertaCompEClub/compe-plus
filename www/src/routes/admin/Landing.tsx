import { useAuth0 } from '@auth0/auth0-react';
import {
    Button,
    Checkbox,
    CircularProgress,
    FormControl,
    FormControlLabel,
    FormGroup,
    Grid,
    makeStyles,
    Paper,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
    Typography,
    withStyles,
} from '@material-ui/core';
import React, { FC, useEffect, useState } from 'react';

import { useAdminDispatch, useAdminSelector } from '../../redux/substores/admin/adminHooks';
import getAllUserRoles from '../../redux/substores/admin/thunks/getAllUserRoles';
import useGlobalStyles from '../../styles/style';
import { UserRole } from '../../util/serverResponses';

const StyledTableCell = withStyles((theme) => ({
    head: {
        backgroundColor: theme.palette.primary.dark,
        color: theme.palette.common.white,
        fontSize: 18,
    },
}))(TableCell);

const StyledTableRow = withStyles((theme) => ({
    root: {
        '&:nth-of-type(odd)': {
            backgroundColor: theme.palette.secondary.main,
        },
    },
}))(TableRow);

interface UserTableProps {
    userRoles: userRoleTableEntries;
    action: (id: string, isUnassigned: boolean, role: string) => void;
}

const UserRoleTable: FC<UserTableProps> = (props: UserTableProps) => {
    const classes = useStyles();
    return (
        <TableContainer component={Paper}>
            <Table className={classes.table} aria-label='simple table'>
                <TableHead>
                    <StyledTableRow>
                        <StyledTableCell>Name</StyledTableCell>
                        <StyledTableCell align='right'>Roles</StyledTableCell>
                    </StyledTableRow>
                </TableHead>
                <TableBody>
                    {Object.entries(props.userRoles).map(([id, userRole]) => {
                        return (
                            <StyledTableRow key={id}>
                                <StyledTableCell component='th' scope='row'>
                                    {userRole.name}
                                </StyledTableCell>
                                <StyledTableCell align='right'>
                                    <FormControl>
                                        <FormGroup>
                                            <>
                                                <FormControlLabel
                                                    control={
                                                        <Checkbox
                                                            checked={userRole.roles.includes('student')}
                                                            onChange={(e) => props.action(id, !e.target.checked, e.target.name)}
                                                            name='student'
                                                            color='primary'
                                                        />
                                                    }
                                                    label='Student'
                                                />
                                                <FormControlLabel
                                                    control={
                                                        <Checkbox
                                                            checked={userRole.roles.includes('reviewer')}
                                                            onChange={(e) => props.action(id, !e.target.checked, e.target.name)}
                                                            name='reviewer'
                                                            color='primary'
                                                        />
                                                    }
                                                    label='Reviewer'
                                                />
                                                <FormControlLabel
                                                    control={
                                                        <Checkbox
                                                            checked={userRole.roles.includes('interviewer')}
                                                            onChange={(e) => props.action(id, !e.target.checked, e.target.name)}
                                                            name='interviewer'
                                                            color='primary'
                                                        />
                                                    }
                                                    label='Interviewer'
                                                />
                                                <FormControlLabel
                                                    control={
                                                        <Checkbox
                                                            checked={userRole.roles.includes('admin')}
                                                            onChange={(e) => props.action(id, !e.target.checked, e.target.name)}
                                                            name='admin'
                                                            color='primary'
                                                        />
                                                    }
                                                    label='Admin'
                                                />
                                            </>
                                        </FormGroup>
                                    </FormControl>
                                </StyledTableCell>
                            </StyledTableRow>
                        );
                    })}
                </TableBody>
            </Table>
        </TableContainer>
    );
};

type userRoleTableEntries = {
    [key: string]: {
        roles: string[];
        name: string;
    };
};

type SuperSet = {
    [key: string]: number;
};

const initialUserRoleTableState: userRoleTableEntries = {};
const initialUpdatedUserRolesState: string[] = [];

// https://stackoverflow.com/questions/6229197/how-to-know-if-two-arrays-have-the-same-values/55614659#55614659
function areArraysEqualSets(a1: string[], a2: string[]) {
    const superSet: SuperSet = {};
    for (const i of a1) {
        const e = i + typeof i;
        superSet[e] = 1;
    }

    for (const i of a2) {
        const e = i + typeof i;
        if (!superSet[e]) {
            return false;
        }
        superSet[e] = 2;
    }

    for (const e in superSet) {
        if (superSet[e] === 1) {
            return false;
        }
    }

    return true;
}

const Landing: FC = () => {
    const classes = useStyles();
    const globalClasses = useGlobalStyles();
    const { userRoles, userRolesIsLoading, shouldReload } = useAdminSelector((state) => state.userRoles);
    const { getAccessTokenSilently } = useAuth0();
    const dispatch = useAdminDispatch();

    const [userRolesTable, setUserRolesTable] = useState(initialUserRoleTableState);
    const [originalUserRoles, setOriginalUserRoles] = useState(initialUserRoleTableState);
    const [updatedUserRoles, setUpdatedUserRoles] = useState(initialUpdatedUserRolesState);

    const createUserRolesTable = (userRoles: UserRole[]) => {
        return userRoles.reduce((accum, userRole) => ({ ...accum, [userRole.id]: { roles: userRole.roles, name: userRole.fullName } }), {});
    };

    const updateRole = (id: string, isUnassigned: boolean, role: string) => {
        const updatedRoles = isUnassigned ? userRolesTable[id].roles.filter((userRole) => userRole !== role) : [...userRolesTable[id].roles, role];

        if (areArraysEqualSets(originalUserRoles[id].roles, updatedRoles)) {
            setUpdatedUserRoles(updatedUserRoles.filter((userId) => userId !== id));
        } else if (!updatedUserRoles.includes(id)) {
            setUpdatedUserRoles([...updatedUserRoles, id]);
        }

        setUserRolesTable({ ...userRolesTable, [id]: { ...userRolesTable[id], roles: updatedRoles } });
    };

    useEffect(() => {
        if (shouldReload) {
            dispatch(getAllUserRoles({ tokenAcquirer: getAccessTokenSilently }));
        }
    }, [shouldReload]);

    useEffect(() => {
        if (userRoles.length > 0) {
            setUserRolesTable(createUserRolesTable(userRoles));
            setOriginalUserRoles(createUserRolesTable(userRoles));
        }
    }, [userRoles]);

    return (
        <Grid container justify='center' alignItems='center' direction='column' className={classes.pageGrid}>
            <Grid container justify='flex-start' alignItems='flex-start' direction='column'>
                <Typography variant='h1'>Users</Typography>
                <Button disabled={updatedUserRoles.length <= 0} className={globalClasses.main_button}>
                    Update Roles
                </Button>
                <Grid container justify='flex-start' alignItems='flex-start' className={classes.wrapper}>
                    {userRolesIsLoading && (
                        <Grid container justify='center' alignItems='flex-start'>
                            <CircularProgress />
                        </Grid>
                    )}
                    {!userRolesIsLoading && userRoles.length == 0 && <Typography>There are no users in CompE+, something is wrong :(</Typography>}
                    {!userRolesIsLoading && userRoles.length > 0 && <UserRoleTable userRoles={userRolesTable} action={updateRole} />}
                </Grid>
            </Grid>
        </Grid>
    );
};

const useStyles = makeStyles({
    table: {
        minWidth: 650,
    },
    wrapper: {
        marginTop: 25,
        marginBottom: 25,
    },
    pageGrid: {
        padding: 50,
    },
});

export default Landing;
