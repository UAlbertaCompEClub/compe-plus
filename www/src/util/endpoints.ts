import config from './config';

export const getMe = `${config.server.endpoint}/api/v1/users/me`;
export const getResumeReviews = `${config.server.endpoint}/api/v1/resume-reviews`;
export const getUserRole = (userId: string): string => `${config.server.endpoint}/api/v1/users/${encodeURIComponent(userId)}/roles`;

export const postUsers = `${config.server.endpoint}/api/v1/users`;

export const postResumeReviews = `${config.server.endpoint}/api/v1/resume-reviews`;
export const patchResumeReviews = (resumeReviewId: string): string => `${config.server.endpoint}/api/v1/resume-reviews/${resumeReviewId}`;
export const patchMyResumeReview = (resumeReviewId: string): string => `${config.server.endpoint}/api/v1/resume-reviews/${resumeReviewId}`;

export const postDocuments = (resumeReviewId: string): string => `${config.server.endpoint}/api/v1/resume-reviews/${resumeReviewId}/documents`;
export const getMyDocuments = (resumeReviewId: string): string => `${config.server.endpoint}/api/v1/resume-reviews/${resumeReviewId}/documents`;
