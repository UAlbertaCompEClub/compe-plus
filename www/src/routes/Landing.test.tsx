import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import Landing, { Intro } from './Landing';
import { shallowWithAuth0 } from '../util/testWithAuth0';

Enzyme.configure({ adapter: new Adapter() });
describe('Landing', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Landing />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Intro', () => {
    it.each`
        isAuthenticated | friendlyName
        ${true}         | ${'when authenticated'}
        ${false}        | ${'when unauthenticated'}
    `('renders correctly', ({ isAuthenticated, friendlyName }) => {
        const wrapper = shallowWithAuth0(<Intro />, {
            isAuthenticated,
        });
        expect(wrapper).toMatchSnapshot(friendlyName);
    });
});
