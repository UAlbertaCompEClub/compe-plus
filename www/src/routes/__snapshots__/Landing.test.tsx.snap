// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders correctly: when authenticated 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  item={true}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      xs={5}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)
          maxWidth="sm"
        >
          <img
            className="makeStyles-logo_1-5"
            src="logo_black.svg"
          />
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Typography))
          variant="h1"
        >
          CompE+
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          Practice, Insight, and Support
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={7}
    >
      <Styled(MuiBox)
        maxWidth="sm"
      >
        <img
          className="makeStyles-svg-1"
          src="pair_programming.svg"
        />
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      justify="center"
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        direction="column"
        item={true}
        justify="center"
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            component="h2"
            variant="h2"
          >
            CompE+ is a student-led pilot program to provide support for computer engineering students looking for internships
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Button))
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            Get Started
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    spacing={10}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={8}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          component="h2"
          noWrap={true}
          variant="h1"
        >
          What’s CompE+ ?
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          component="h2"
          variant="body1"
        >
          CompE+ is a student led pilot program to provide CompE students with practice, insight, and support when finding internships in the tech industry. Arguably, computer engineering is a unique discipline and requires different resources and support - that’s where CompE+ comes in!
           
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={4}
    >
      <img
        className="makeStyles-logo_2-4"
        src="logo_black.svg"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Intro renders correctly: when unauthenticated 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  item={true}
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      xs={5}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)
          maxWidth="sm"
        >
          <img
            className="makeStyles-logo_1-5"
            src="logo_black.svg"
          />
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Typography))
          variant="h1"
        >
          CompE+
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          Practice, Insight, and Support
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={7}
    >
      <Styled(MuiBox)
        maxWidth="sm"
      >
        <img
          className="makeStyles-svg-1"
          src="pair_programming.svg"
        />
      </Styled(MuiBox)>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      item={true}
      justify="center"
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        direction="column"
        item={true}
        justify="center"
        spacing={3}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Typography))
            align="center"
            component="h2"
            variant="h2"
          >
            CompE+ is a student-led pilot program to provide support for computer engineering students looking for internships
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Button))
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            Get Started
          </WithStyles(ForwardRef(Button))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    spacing={10}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={8}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          component="h2"
          noWrap={true}
          variant="h1"
        >
          What’s CompE+ ?
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          component="h2"
          variant="body1"
        >
          CompE+ is a student led pilot program to provide CompE students with practice, insight, and support when finding internships in the tech industry. Arguably, computer engineering is a unique discipline and requires different resources and support - that’s where CompE+ comes in!
           
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={4}
    >
      <img
        className="makeStyles-logo_2-4"
        src="logo_black.svg"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Landing renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="row"
  justify="center"
  style={
    Object {
      "backgroundImage": "url(LandingBackground.svg)",
      "backgroundSize": "cover",
      "paddingBottom": 80,
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    spacing={10}
    xs={10}
  >
    <Intro />
    <Values />
    <Services />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
